src/main/java/com/google/research/bleth/simulator/RandomMovementStrategy.java

```diff
@@ -0,0 +1,36 @@
+package com.google.research.bleth.simulator;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/** A random movement strategy for an agent. */
+public class RandomMovementStrategy implements MovementStrategy {
+
+    private final static List<List<Integer>> directions =
+            ImmutableList.of(ImmutableList.of(0, -1), ImmutableList.of(0, 1),
+                             ImmutableList.of(-1, 0), ImmutableList.of(1, 0));
+
+    /**
+     * Determine and return the new location of a randomly moving agent, which is walking a single step in one of the
+     * possible directions from its current location. If an agent can't move to any direction, return its current location.
+     * @param board is the board that the agent is placed on.
+     * @param currentLocation is the current location of the agent on the board.
+     * @return the location on the board that the agent is moving to.
+     */
+    @Override
+    public Location moveTo(Board board, Location currentLocation) {
+        List<List<Integer>> shuffledDirections = new ArrayList<>(directions);
+        Collections.shuffle(shuffledDirections);
+        for (List<Integer> direction : shuffledDirections) {
+            Location newLocation = new Location(currentLocation.row + direction.get(0),
```

Comment:
I assume this is not the only situation where we would like to add one location to another in order to generate a new location.

I wanted to suggest we use operators overloading to overload the '+' operator so we can just use:
```java
Location newLocation = currentLocation + direction;
```
But it seems like Java doesn't support operator overloading. 
Alternatively we can write a static 'add' method, and then use:
```java
Location newLocation = Location::Add(currentLocation, direction);
```

Reply:
Okay I just noticed that 'direction' is not a Location but just a list of two integers. So we can either define a Direction class and then write a method to move a Location in a certain Direction, or just leave it as is.